### PR TITLE
feat: implement storage update handler

### PR DESCRIPTION
**Pull request by aleacevedo**
# feat: implement storage update handler

## Summary

Every time a new version of PlugController is deployed, controller gonna check if some update should be done on the storage

## Shourtcut
https://app.shortcut.com/terminalsystems/story/24864/refactor-version-storage-schema

### CODE REVIEW

File: src/utils/storage/utils.ts [hunk 1]
```diff
@@ -19,3 +20,52 @@ export const checkForError = (): Error | undefined => {
...
+  if (index === VERSION_PATH.length)
+    return storage;
+
+  let newStorage = storage;
+  for (const version in VERSION_PATH.slice(index)) {
+    newStorage = VERSION_HANDLER[version](storage);
+  }
+
+  return newStorage;
+}
```
Comment (mattgle) [inline src/utils/storage/utils.ts:71]: edd eof

File: src/PlugKeyRing/index.ts [hunk 2]
```diff
@@ -157,6 +162,10 @@ class PlugKeyRing {
       this.state = { ...decrypted, wallets };
       this.isInitialized = isInitialized;
       this.currentWalletId = currentWalletId;
+      if (newVersion !== version) {
+        this.saveEncryptedState({ wallets }, password);
```
Comment (tomiir) [inline src/PlugKeyRing/index.ts:166]: why do this?

File: src/utils/storage/utils.ts [hunk 3]
```diff
@@ -19,3 +20,52 @@ export const checkForError = (): Error | undefined => {
   // repair incomplete error object (eg chromium v77)
   return new Error(lastError.message);
 };
+
+const VERSION_PATH: Array<string> = []
+
+const VERSION_HANDLER: { [version: string]: (storage: any) => PlugState } = {};
+
+const compareVersion = (a: string, b: string) => {
```
Comment (tomiir) [inline src/utils/storage/utils.ts:28]: add return types

File: src/utils/storage/utils.ts [hunk 4]
```diff
@@ -19,3 +20,52 @@ export const checkForError = (): Error | undefined => {
...
+  const arrA = a.split('.');
+  const arrB = b.split('.');
+  if (arrA.length !== 3 || arrB.length !== 3)
+    throw Error('Storage Hande Update: invalid version');
+  for (let index = 0; index < 3; index++) {
+    const numbA = parseInt(arrA[index])
+    const numbB = parseInt(arrB[index])
+    if (numbA > numbB)
+      return -1;
```
Comment (tomiir) [inline src/utils/storage/utils.ts:37]: why return numbers?

File: src/utils/storage/utils.ts [hunk 5]
```diff
@@ -19,3 +20,52 @@ export const checkForError = (): Error | undefined => {
...
+
+
+export const handleStorageUpdate = (storageVersion: string | undefined, storage: any): PlugState => {
+  const index = getVersionIndex(storageVersion);
+  if (index === VERSION_PATH.length)
+    return storage;
+
+  let newStorage = storage;
+  for (const version in VERSION_PATH.slice(index)) {
+    newStorage = VERSION_HANDLER[version](storage);
```
Comment (tomiir) [inline src/utils/storage/utils.ts:67]: I don't see where `VERSION_HANDLER` is filled with info 🤔

**Review by tomiir — APPROVED** [review]

**Pull request merged**